### PR TITLE
HUDS! (suit sensors improvement) 

### PR DESCRIPTION
**Pull request by Bod9001**
adds HUDs have a look yourself
image

GIF297

### Changelog:

CL: [New] Adds Security Hud
CL: [New] Adds Medical Hud
CL: [Improvement] grab intent will now grab stuff when clicking
CL: [Fix] hopefully fixes the suit sensors

### CODE REVIEW

File: UnityProject/Assets/Scripts/Clothing/Eyewear/HUDItemBase.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	{
+		pickupable =  GetComponent<Pickupable>();
+	}
+
+	#region InventoryMove
+
+	public RegisterPlayer CurrentlyOn { get; set; }
+	bool IItemInOutMovedPlayer.PreviousSetValid { get; set; }
+
+	void IItemInOutMovedPlayer.ChangingPlayer(RegisterPlayer HideForPlayer, RegisterPlayer ShowForPlayer)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Clothing/Eyewear/HUDItemBase.cs:20]: `private void`

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Clothing/Eyewear/HUDItemBase.cs:20]: also dont put functions in the middle of variables

File: UnityProject/Assets/Scripts/Clothing/Eyewear/MedicalHUD.cs [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+		Critical, //flashing
+		Defibrillatorble, //that defibrillator one
+		NoSoul, //Skull, you are Dead! no, you! Pow hAhA. You are dead, no big surprise
+		Ill, //green
+		XenoEgg //eggy //TODO
+
+	}
+
+
+	public enum HealthBarPercentage
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Clothing/Eyewear/MedicalHUD.cs:212]: why

File: UnityProject/Assets/Scripts/Clothing/Eyewear/SecurityHUD.cs [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+			case JobType.WARDEN:
+				return JobIcon.Warden;
+			default:
+				return JobIcon.Unknown;
+		}
+	}
+
+
+	public void Awake()
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Clothing/Eyewear/SecurityHUD.cs]: put unity functions at the top of functions

**Review by MaxIsJoe — CHANGES_REQUESTED** [review]
Looks good, just some minor changes needed.

File: UnityProject/Assets/Scripts/Clothing/Eyewear/HUDItemBase.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Clothing/Eyewear/HUDItemBase.cs:20]: Is automatically private, you can't access it when accessing the HUDItemBase class

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Clothing/Eyewear/HUDItemBase.cs:20]: though if you cast to IItemInOutMovedPlayer it is then visible

File: UnityProject/Assets/Scripts/Clothing/Eyewear/MedicalHUD.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Clothing/Eyewear/MedicalHUD.cs:212]: it's what the sprites are

File: UnityProject/Assets/Scripts/Clothing/Eyewear/SecurityHUD.cs (hunk 3 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Clothing/Eyewear/SecurityHUD.cs]: sounds good though do we have anywhere where we have example layouts?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Clothing/Eyewear/SecurityHUD.cs]: I'll update the code guidelines to point that out.

**Review by MaxIsJoe — APPROVED** [review]

**Pull request merged**